syntax = "proto3";
package process_definition;




import "flowable_sdk/model/flowable/process_definition.proto";

/*
ListProcessDefinition请求
*/
message ListProcessDefinitionRequest {
    /*
    起始个数
    */
    int32 start = 1 ;
    /*
    页码大小
    */
    int32 size = 2 ;
    /*
    排序字段
    */
    string sort = 3 ;
    /*
    排序顺序
    */
    string order = 4 ;
    /*
    版本
    */
    int32 version = 5 ;
    /*
    名称
    */
    string name = 6 ;
    /*
    名称模糊查询
    */
    string nameLike = 7 ;
    /*
    key
    */
    string key = 8 ;
    /*
    key模糊查询
    */
    string keyLike = 9 ;
    /*
    资源名称
    */
    string resourceName = 10 ;
    /*
    资源名称模糊搜索
    */
    string resourceNameLike = 11 ;
    /*
    分类
    */
    string category = 12 ;
    /*
    分类模糊搜索
    */
    string categoryLike = 13 ;
    /*
    分类不等于
    */
    string categoryNotEquals = 14 ;
    /*
    部署ID
    */
    string deploymentId = 15 ;
    /*
    启动用户
    */
    string startableByUser = 16 ;
    /*
    是否只返回最新版
    */
    string lastest = 17 ;
    /*
    是否挂起
    */
    string suspended = 18 ;
    /*
    querystring指定传的field
    */
    repeated string XXX_RestFieldMask = 19 ;
}

/*
ListProcessDefinition返回
*/
message ListProcessDefinitionResponse {
    /*
    定义列表
    */
    repeated flowable.FlowableProcessDefinition data = 1 ;
    /*
    总数
    */
    int32 total = 2 ;
    /*
    起始个数
    */
    int32 start = 3 ;
    /*
    排序字段
    */
    string sort = 4 ;
    /*
    排序顺序
    */
    string order = 5 ;
    /*
    一页大小
    */
    int32 size = 6 ;
}

/*
ListProcessDefinitionApi返回
*/
message ListProcessDefinitionResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    process_definition.ListProcessDefinitionResponse data = 4 ;
}

